Entities/EffortReport.cs

```diff
@@ -1,10 +1,47 @@
-﻿namespace Payroll.Entities
+﻿using Payroll.Application.Enums;
+
+namespace Payroll.Entities
 {
     public class EffortReport
     {
+        /// <summary>
+        /// Unique identifier of the employee.
+        /// </summary>
         public Guid EmployeeId { get; set; }
-        /*
-         * Write some code here.
-         */
+        /// <summary>
+        /// Effort time made by the employee for the reporting period.
+        /// </summary>
+        public decimal EffortTime { get; set; }
+        /// <summary>
+        /// Payment rate at which the employee is paid.
+        /// </summary>
+        public decimal Payment { get; set; }
+        /// <summary>
+        /// Type of employees.
+        /// </summary>
+        public EmployeesType EmployeeType { get; set; }
+        /// <summary>
+        /// Salary of the employee.
+        /// </summary>
+        public decimal Salary
+        {
+            get
+            {
+                return Payment * EffortTime;
+            }
+        }
+/// <summary>
+/// Calculate the payment rate.
+/// </summary>
+/// <returns></returns>
+        public string CalculatePaymentRate()
+        {
+            return $"{Payment} {(EmployeeType == EmployeesType.Hourly ? "$ / hour" : "$ / month")}";
```

Comment:
What happens to these two methods if we decide to add a couple of other employment types?
We'll have to add more logic in this class (breaking Open-Closed rule).

May subtyping and polymorphism be used instead?

Reply:
Made changes to follow Open-Closed principle.